lib/buildFactory.js

```diff
@@ -263,8 +263,9 @@ class BuildFactory extends BaseFactory {
 
                     const build = await super.create(modelConfig);
 
-                    await steps.reduce((p, step) =>
-                        p.then(() => stepFactory.create(Object.assign(
+                    await steps.reduce(async (p, step) =>
+                        // eslint-disable-next-line no-return-await
+                        await p.then(() => stepFactory.create(Object.assign(
```

Comment:
Don't we have to `await` stepFactory.create() ?

Reply:
perhaps switch to `for of` for ensuring order. We do something similar here https://github.com/screwdriver-cd/screwdriver/pull/2011/files#diff-7a514e440aa937beee722c7ac84dfb93R1263-R1271

Reply:
@jithine 
It is a similar case.
Modify it using the `for of`.